File: server/models/Category.js
const mongoose = require('mongoose');

const CategorySchema = new mongoose.Schema(
  {
    //id will be set by default
    name: {
      type: mongoose.Schema.Types.String,
      required: true,
    },
    image: {
      type: mongoose.Schema.Types.String,
      required: true,
    },
    productList: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Product',
      },
    ],
  },
  {
    writeConcern: {
      w: 'majority',
      j: true,
      wtimeout: 1000,
    },
  }
);

const Category = mongoose.model('Category', CategorySchema);

module.exports = { Category, CategorySchema };

File: application/RaneenXamarinProject/RaneenXamarinProject/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RaneenXamarinProject.Models
{
    public class Category
    {
        public string id { get; set; }
        public string name { get; set; }
        public string image { get; set; }
        public List<Product1> product_list { get; set; }
    }
}
